File: C++/matrix-diagonal-sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int sum = 0;
        int sz = mat.size();
        for (int i = 0; i < sz; i++) {
            int j = sz - i - 1;
            sum += mat[i][i];
            if (i != j) {
                sum += mat[i][j];
            }
        }
        return sum;
    }
};

File: Kotlin/matrix-diagonal-sum.kt
class Solution {
    fun diagonalSum(mat: Array<IntArray>): Int {
        val n = mat.size
        var sum = 0
        for (i in 0 until n) {
            sum += mat[i][i] + mat[i][n - 1 - i]
        }
        if (n % 2 == 1)
            sum -= mat[n / 2][n / 2]        
        return sum
    }
}